tests/epics/signal/test_signals.py

```diff
@@ -886,3 +886,27 @@ async def test_signal_not_return_no_limits(ioc: IOC):
     await sig.connect()
     datakey = (await sig.describe())[""]
     assert not hasattr(datakey, "limits")
+
+
+async def test_signals_created_for_prec_0_float_can_use_int(ioc: IOC):
+    if ioc.protocol == "pva":
+        pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:float_prec_0"
+        sig = epics_signal_rw(int, pv_name)
+        await sig.connect()
+    elif ioc.protocol == "ca":
+        pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:float_prec_0"
+        sig = epics_signal_rw(int, pv_name)
+        await sig.connect()
+
+
+async def test_signals_created_for_not_prec_0_float_cannot_use_int(ioc: IOC):
+    if ioc.protocol == "pva":
+        pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:float_prec_1"
+        sig = epics_signal_rw(int, pv_name)
+        with pytest.raises(TypeError):
+            await sig.connect()
+    elif ioc.protocol == "ca":
+        pv_name = f"{ioc.protocol}://{PV_PREFIX}:{ioc.protocol}:float_prec_1"
+        sig = epics_signal_rw(int, pv_name)
+        with pytest.raises(TypeError):
+            await sig.connect()
```

Comment:
What's the purpose of `if ioc.protocol == "..."` here? The protocol in use within the ioc is always going to be one or the other here?

Even if it is variable between runs of tests (it really shouldn't be), the code in each clause is the same.

Reply:
ah yeah, no need for the if's